app/src/main/java/movierating/services/ReviewServiceImpl1.java

```diff
@@ -15,7 +18,21 @@ public ReviewServiceImpl1(UserRepository userRepository, MovieRepository movieRe
 
 
     @Override
-    public void addReview() {
+    public void addReview(String userName, String movieName, int rating) {
+
+        User user = this.userRepository.getUserList().get(userName);
+        if (user == null) {
+            System.out.println("He is not a user of this Service");
+            return;
+        }
+        Review review = new Review(user, rating);
+        Movie movie = this.movieRepository.getMovieList().get(movieName);
+        if (movie == null) {
+            System.out.println("Movie is not released yet");
+            return;
+        }
+
+        movie.getReviewList().add(review);
```

Comment:
Should the review count of an user be incremented here?

Reply:
Yes. done.